pkg/discovery/dns/provider.go

```diff
@@ -143,6 +147,11 @@ func (p *Provider) Resolve(ctx context.Context, addrs []string) {
 	p.resolverAddrs.Submit()
 
 	p.resolved = resolvedAddrs
+	if len(errs) > 0 {
```

Comment:
Power tip: we can do just returns. `errs.Err()` (:  Without check! (check will be done on err)

Reply:
Ohh great, let me add it up!